util.go

```diff
@@ -0,0 +1,219 @@
+// state.go - Katzenpost non-voting authority server state.
+// Copyright (C) 2017  Yawning Angel.
+
+package katzenmint
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/cosmos/iavl"
+	"github.com/hashcloak/katzenmint-pki/s11n"
+	"github.com/katzenpost/core/crypto/rand"
+	"github.com/katzenpost/core/epochtime"
+	"github.com/katzenpost/core/pki"
+	"github.com/katzenpost/core/sphinx/constants"
+	"github.com/tendermint/tendermint/crypto/merkle"
+)
+
+const (
+	descriptorsBucket = "k_descriptors"
+	documentsBucket   = "k_documents"
+	authoritiesBucket = "k_authorities"
+)
+
+func storageKey(keyPrefix string, keyID []byte, version uint64) (key []byte) {
+	verHex := make([]byte, 8)
+	binary.PutUvarint(verHex, version)
+	verHex = []byte(EncodeHex(verHex))
+	IDHex := []byte(EncodeHex(keyID))
+
+	key = make([]byte, len(keyPrefix))
+	copy(key, keyPrefix)
+	key = append(key[:], ':')
+	key = append(key[:], verHex[:]...)
+	key = append(key[:], ':')
+	key = append(key[:], IDHex[:]...)
+	return
+}
+
+func (state *KatzenmintState) documentForEpoch(epoch uint64) ([]byte, merkle.ProofOperator, error) {
```

Comment:
IMHO we should put this function into state.go, If we only use this function in katzenmint state.